pkg/apis/extensions/types.go

```diff
@@ -809,3 +810,75 @@ type PodSecurityPolicyList struct {
 
 	Items []PodSecurityPolicy `json:"items"`
 }
+
+// +genclient=true
+
+// Template contains one or more partially complete configs with substitution parameters.
+type Template struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"`
+
+	Spec TemplateSpec `json:"spec"`
+}
+
+type TemplateSpec struct {
+	// Optional: Parameters is an array of Parameters used during the
+	// Template to Config transformation.
+	Parameters []Parameter `json:"parameters,omitempty"`
+
+	// Required: A yaml config list of resources to create
+	Objects []runtime.RawExtension `json:"objects"`
+}
```

Comment:
There are places where the parameter would be used as both string and
number - for instance, REPLICAS used on both the "replicas" field of a
PetSet and as the value of environment variable REPLICA_COUNT. There
is no guarantee the server knows (or can know) the fields of the
objects stored in a template.  Processing of the template needs to be
done without casting to a go type, because it should be possible to
store a template on a server for which some fields are not known to
that server without losing those fields (fields added later to API
v1).